### PR TITLE
[subset] proposed update to subset set apis.

### PR DESCRIPTION
**Pull request by garretrieger**
For #3130

### CODE REVIEW

File: src/hb-subset.h [hunk 1]
```diff
@@ -84,6 +84,29 @@ typedef enum { /*< flags >*/
...
+ * List of sets that can be configured on the subset input.
+ *
+ * Since: REPLACE
+ **/
+typedef enum { /*< flags >*/
+  HB_SUBSET_SETS_LAYOUT_FEATURE_TAG = 0,
+  HB_SUBSET_SETS_NAME_INDEX,
+  HB_SUBSET_SETS_NAME_LANG_INDEX,
+  HB_SUBSET_SETS_NO_SUBSET_TABLE_TAG,
+  HB_SUBSET_SETS_DROP_TABLE_TAG,
```
Comment (behdad) [inline src/hb-subset.h:107]: Let's add a `HB_SUBSET_SETS_COUNT` at the end? Similarly probably add `HB_SUBSET_FLAG_DEFINED` as or of all.

File: src/hb-subset.h [hunk 2]
```diff
@@ -84,6 +84,29 @@ typedef enum { /*< flags >*/
...
+ * subsetted.
+ * @HB_SUBSET_SETS_DROP_TABLE_TAG: the set of table tags which specifies tables which will be dropped
+ * in the subset.
+ *
+ * List of sets that can be configured on the subset input.
+ *
+ * Since: REPLACE
+ **/
+typedef enum { /*< flags >*/
+  HB_SUBSET_SETS_LAYOUT_FEATURE_TAG = 0,
```
Comment (behdad) [inline src/hb-subset.h:103]: Make 0 be "UNICODE" and 1 be "GLYPH_INDEX".

**Comment by bungeman** [comment]
Already commented over on the API proposal itself, but the implementation of this is going to look like a switch, or worse these sets will get placed into an array and be accessed by enum. But to what end is this extra level of indirection? What is the improvement? Is it fewer lines in the documentation? That's just a documentation formatting issue. Fewer overall API functions? But there is no actual reduction in complexity here. In fact, it's really confusing that a bunch of unrelated data happens to be going through one API function simply because they accidentally have the same type. Also, I can detect the existence of a function, but not an enum value, so this is actually a step backwards in API usefulness for users trying to stay near tip of tree.

File: src/hb-subset-input.hh [hunk 3]
```diff
@@ -40,23 +42,110 @@ struct hb_subset_input_t
...
+  {
+    return sizeof (set_ptrs) / sizeof (hb_set_t*);
+  }
+
+  inline hb_array_t<hb_set_t*> sets_iter ()
+  {
+    return hb_array_t<hb_set_t*> (set_ptrs, num_sets ());
+  }
+
+  inline hb_set_t* unicodes()
```
Comment (behdad) [inline src/hb-subset-input.hh:70]: Are these functions really necessary? Instead of just typing `input->sets.unicodes`?

Comment (garretrieger) [reply src/hb-subset-input.hh]: Removed.

File: src/hb-subset.h [hunk 4]
```diff
@@ -79,6 +79,33 @@ typedef enum { /*< flags >*/
...
+ * @HB_SUBSET_SETS_DROP_TABLE_TAG: the set of table tags which specifies tables which will be dropped
+ * in the subset.
+ * @HB_SUBSET_SETS_NAME_ID: the set of name ids that will be retained.
+ * @HB_SUBSET_SETS_NAME_LANG_ID: the set of name lang ids that will be retained.
+ * @HB_SUBSET_SETS_LAYOUT_FEATURE_TAG: the set of layout feature tags that will be retained
+ * in the subset.
+ *
+ * List of sets that can be configured on the subset input.
+ *
+ * Since: REPLACE
```
Comment (behdad) [inline src/hb-subset.h:97]: "REPLACEME" please. We grep for that.

**Pull request merged**